Sources/Towel/Mathematics/Matrix.cs

```diff
@@ -465,6 +464,34 @@ public MatrixElementFraction<T> Abs()
 		}
 #endregion
 
+#region GetDeterminant Gaussian elimination analytically
+		internal static T GetDeterminantNonNumericGaussian(Matrix<T> matrix, int n)
+		{
+			if (n == 1)
+				return matrix.Get(0, 0);
+
+			var elemMatrix = new Matrix<MatrixElementFraction<T>>(n, n, 
+				(x, y) => new MatrixElementFraction<T>(matrix.Get(x, y)));
+			for (int k = 1; k < n; k++)
+				for (int j = k; j < n; j++)
+				{
+					var m = elemMatrix.Get(j, k - 1) / elemMatrix.Get(k - 1, k - 1);
+					for (int i = 0; i < n + 0; i++)
+						elemMatrix.Set(j, i, elemMatrix.Get(j, i) - m * elemMatrix.Get(k - 1, i));
+				}
+
+			var det = new MatrixElementFraction<T>(Constant<T>.One);
+			for (int i = 0; i < n; i++)
+				det *= elemMatrix.Get(i, i);
+
+			return det.Value;
```

Comment:
There should be a check where det's Denominator is 0

Reply:
Hmm but then it won't fully non-numeric. So if you take this method, you might want to add
```cs
if (det.IsDividedByZero)
    return Constant<T>.Zero;
```